lmcache/cache_engine.py

```diff
@@ -348,6 +348,93 @@ def retrieve(
                     f"elapsed time {ed - st}")
         return ret, retrieved_token_count
 
+    @_lmcache_nvtx_annotate
+    @torch.no_grad()
+    def get_locations(
+        self,
+        tokens: torch.Tensor,
+    ) -> List[Optional[List[str]]]:
+        """
+        Checks the locations of KV cache of the tokens from the cache engine.
+        The return should be a list of the locations for each block.
+
+        Input:
+            tokens: the input tokens, with shape [seq_len]
+
+            format is either 'huggingface' or 'vllm'
+                For huggingface, it should have the shape of 
+                [num_heads, num_tokens, head_size]
+                
+                For vllm, it should have the shape of 
+                [num_tokens, num_heads, head_size]
+
+        Output:
+            List[List[locations]]:
+            List of the locations (ex. ['local DRAM']) of storage of each block 
+            The entry will be None if the block is not found 
+            or the last block is not full (ex. 13 token / (16 token per block)) 
+        """
+
+        chunk_hashes = self._prefix_hash(self._chunk_tokens(tokens))
+
+        ret_locations_list: list[Optional[list[str]]] = []
+        for chunk_hash in chunk_hashes:
+            list_locations = self.engine_.where_is(
+                self._make_key(chunk_hash, self.metadata.fmt))
+            if list_locations[0] == 'NOT IN CACHE':
```

Comment:
A better design is to use `enum` class, but this will be a large-scale change so feel free to ignore it for now.

Reply:
+1. Please define an enum class for the locations, maybe in `utils.py`. Both this function and the `where_is` function should use this enum class.

Reply:
Let's fix this in the next version? This requires a lot of code change but has no impact on functionality.